sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala

```diff
@@ -101,12 +104,15 @@ private[sql] object SparkPlanGraph {
         } else {
           subgraph.nodes += node
         }
+        if (name == "ShuffleExchange" || name == "BroadcastExchange") {
```

Comment:
@nongli, just to make sure that your original comment here was addressed: were you worried about this pattern being incomplete if we add a new type of exchange? If that's the case, then the move from `name.endsWith("Exchange")` to this might make things worse.